include/tvm/ir/module.h

```diff
@@ -64,6 +65,8 @@ class IRModuleNode : public Object {
   /* \brief Additional attributes storing meta-data about the module. */
   DictAttrs attrs;
 
+  GlobalVarSupply global_var_supply;
```

Comment:
Given that we can recreate GlobalVarSupply from the current IRModule, and we don't need global var renaming for most of the passes(that only do local edits of functions).

It might make sense to create helper method to generate GlobalVarSupply rather than maintaining as a member of IRModule when needed, to keep the main IRModule data structure simple. 

Doing so would also simplify the serialization part of IRModule, right now the unordered map data structure is not serialized, as a result save_json/load_json of IRModule will no longer work if we make it as a member.

Reply:
> Given that we can recreate GlobalVarSupply from the current IRModule

I'm not sure this is so trivial. A couple of cases to think through:
- right now we still lower per-backend separately, so the NameSupply needs to track names across a collection of IRModules in the compiler
- Suppose we needed to reserve an extern name not mentioned in the IR anywhere. We would surely need to serialize something to track that.

The first case here seems to be the one most at tension with the idea of making NameSupply idempotent. I agree that NameSupply isn't necessarily attached to any given IRModule now, so having a way to access it should mainly be a convenience method and not participate in IRModule serialization at least right now. I'm not so sure it means that it's valid to serialize an IRModule and destroy a TVM instance, then load it back in another TVM instance and expect the NameSupply to be the same.

Reply:
I agree @areusch , both of your pts seems to point that we should bring NameSupply outside IRModule, and have additional functionalities to re-create from multple (collection) of IRModules